src/main/java/org/candlepin/model/EnvironmentCurator.java

```diff
@@ -71,19 +69,88 @@ public List<Consumer> getEnvironmentConsumers(String environmentId) {
             .getResultList();
     }
 
-    public CandlepinQuery<Environment> listForOwner(Owner o) {
-        DetachedCriteria criteria = this.createSecureDetachedCriteria()
-            .add(Restrictions.eq("owner", o));
+    public List<Environment> listForOwnerByType(Owner o, List<String> type) {
```

Comment:
Few things across these methods:
- missing documentation
- missing input validation (nulls and empty values should be safe or fail fast and explicitly)
- avoid single-character variable names in these cases